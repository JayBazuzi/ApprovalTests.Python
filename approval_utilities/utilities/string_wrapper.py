class StringWrapper:
    def __init__(self) -> None:
        self.string = ""

    def append(self, text: str) -> None:
        self.string += text

    def __str__(self) -> str:
        return self.string

    def __repr__(self) -> str:
        return self.string
