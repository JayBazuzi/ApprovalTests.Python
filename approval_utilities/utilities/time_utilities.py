import os

from typing_extensions import ContextManager
from typing import Optional, Type
from types import TracebackType


def use_utc_timezone() -> ContextManager:
    class TimeZoneSwap:
        def __init__(self) -> None:
            self.timezone: Optional[str] = ""

        def __enter__(self) -> None:
            self.timezone = os.environ.get("TZ")
            os.environ["TZ"] = "UTC"

        def __exit__(
            self,
            exc_type: Optional[Type],
            exc_val: Optional[BaseException],
            exc_tb: Optional[TracebackType],
        ) -> bool:
            if self.timezone is None:
                os.environ.pop("TZ")
            else:
                os.environ["TZ"] = self.timezone

    return TimeZoneSwap()
