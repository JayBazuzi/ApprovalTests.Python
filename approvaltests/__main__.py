from .verify import verify_using_commandline_arguments

verify_using_commandline_arguments()