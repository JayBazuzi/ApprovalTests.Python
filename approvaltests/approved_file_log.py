from pathlib import Path

APPROVAL_TESTS_TEMP_DIRECTORY: str = ".approval_tests_temp"


class ApprovedFilesLog:
    @staticmethod
    def clear_log_file() -> None:
        ApprovedFilesLog.get_approved_files_log().write_text("")

    @staticmethod
    def get_approved_files_log() -> Path:
        path = Path("%s/.approved_files.log" % APPROVAL_TESTS_TEMP_DIRECTORY)
        path.parent.mkdir(parents=True, exist_ok=True)

        return path

    @staticmethod
    def log(approved_file):
        with ApprovedFilesLog.get_approved_files_log().open(mode="a") as file:
            file.write(f"{approved_file}\n")


ApprovedFilesLog.clear_log_file()
