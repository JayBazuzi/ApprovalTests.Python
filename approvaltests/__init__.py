"""Approvaltests module."""
from .version import *
from .utilities import *
from .approvals import *
from .asserts import *

__author__ = "Chris Lucian; Llewellyn Falco; Jim Counts; Tim Preece"
__version__ = version_number
