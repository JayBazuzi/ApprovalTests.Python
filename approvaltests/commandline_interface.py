from pathlib import Path

from approvaltests import verify
from build.lib.approvaltests.commandline_interface import parse_arguments
from namer.cli_namer import CliNamer


# simplify the following lines of code from commandline_interface.py

# 1.  Modify so our future-selves would recognize it
# >>> 2.  Make a test
# 3.  Refactor parse_arguments to be part of a class (suggested by Jay)

# NEw code...
# received = args.received or stdin.read()
# Old code...
# received = args.received
# if args.received is None:
#    received = stdin.read()
# return (args.id, received)



def verify_using_commandline_arguments():
    test_id, received = parse_arguments()
    verify_with_id(received, test_id)


def verify_with_id(received, test_id):
    verify(received, namer=CliNamer(test_id=test_id))
    print(f"Test Passed: {test_id}")


if __name__ == "__main__":
    verify_using_commandline_arguments()


def list_approved_files_in_test_directory(test_dir: Path):
    return [file.name for file in test_dir.iterdir() if file.name.endswith(".approved.txt")]
