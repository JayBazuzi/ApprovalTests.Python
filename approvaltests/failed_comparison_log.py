from pathlib import Path

from approvaltests.approved_file_log import APPROVAL_TESTS_TEMP_DIRECTORY


class FailedComparisonLog:
    @staticmethod
    def clear_log_file() -> None:
        FailedComparisonLog.get_failed_comparison_log().write_text("")

    @staticmethod
    def get_failed_comparison_log() -> Path:
        path = Path("%s/.failed_comparison.log" % APPROVAL_TESTS_TEMP_DIRECTORY)
        path.parent.mkdir(parents=True, exist_ok=True)

        return path

    @staticmethod
    def log(received_file, approved_file):
        with FailedComparisonLog.get_failed_comparison_log().open(mode="a") as file:
            file.write(f"{received_file} -> {approved_file}\n")


FailedComparisonLog.clear_log_file()
