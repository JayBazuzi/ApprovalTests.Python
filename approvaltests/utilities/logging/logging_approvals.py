from typing import Optional, ContextManager, Type
from types import TracebackType

from approvaltests import Options, verify
from testfixtures import LogCapture

from approvaltests.scrubbers.date_scrubber import DateScrubber


def verify_logging(
    *,  # enforce keyword arguments - https://www.python.org/dev/peps/pep-3102/
    options: Optional[Options] = None
) -> ContextManager:
    class VerifyLogging:
        def __init__(self) -> None:
            self.l = LogCapture()
            self.output = "anything"
            self.options = options if options else Options()

        def __enter__(self) -> None:
            self.l.__enter__()

            pass

        def __exit__(
            self,
            exc_type: Optional[Type[BaseException]],
            exc_val: Optional[BaseException],
            exc_tb: Optional[TracebackType],
        ) -> bool:
            self.l.__exit__(exc_type, exc_val, exc_tb)
            self.options = self.options.add_scrubber(
                DateScrubber.get_scrubber_for("2023-07-16 17:39:03.293919")
            )
            verify(self.l, options=self.options)

    return VerifyLogging()
