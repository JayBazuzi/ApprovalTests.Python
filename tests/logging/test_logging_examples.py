import logging

from approval_utilities.utilities.multiline_string_utils import remove_indentation_from
from approvaltests import verify
from approvaltests.namer import NamerFactory
from approvaltests.utilities.logging.logging_approvals import verify_logging


def load_person() -> str:
    logging.info("connecting to the database")
    logging.info("querying a table")
    logging.info("closing the database")

    return remove_indentation_from(
        """
    {
        "first_name":"Britney",
        "last_name": "Spears",
        "profession":"Singer"    
    }
    """
    )


# begin-snippet: test_logging_separately
def test_load_person_logs():
    with verify_logging():
        load_person()


# end-snippet


# begin-snippet: test_logging_separately_results
def test_load_person_results():
    verify(load_person())


# end-snippet


# begin-snippet: testing_logging_combined_with_results
def test_load_person_logs_and_results():
    with verify_logging():
        logging.info(f"result = {load_person()}")


# end-snippet


# begin-snippet: testing_logging_with_namer_factory
def test_load_person_logs_and_results_separately():
    with verify_logging(options=NamerFactory.with_parameters("logging")):
        verify(load_person())


# end-snippet
