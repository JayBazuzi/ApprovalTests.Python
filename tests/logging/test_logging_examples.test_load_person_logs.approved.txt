root INFO
  connecting to the database
root INFO
  querying a table
root INFO
  closing the database
